### DISCUSSION TITLE
SCD pipeline for speaker change reporting in real time

### DISCUSSION THREAD

**Discussion by Yagna24** [pyannote/pyannote-audio#898] [Q&A] [answered]
Hi, I was wondering if there's a possibility to output the speaker change time stamps while processing the audio. 

For example, 
If there's a 4 minute audio and there's a speaker change at Minute 1 i.e. Interval[0,1], is it possible to fetch output or print the start and end timestamps while it goes to the next speaker segment i.e. from Minute 1 to Minute 2 per say (1,2]. 

I would extremely appreciate your help.

**Comment by hbredin** [comment, marked as answer]
As of  today (Feb. 2022), real-time processing is not supported in pyannote.
I recommend you use @juanmc2005's [`diart`](https://github.com/juanmc2005/StreamingSpeakerDiarization/) toolkit for this purpose.

**Reply by Yagna24** [reply]
Thank you for the response.

**Reply by Anticope12** [reply]
Hi @hbredin , Just reopening this question If there is a plan to process real-time processing in pyannote?

**Reply by hbredin** [reply]
* Short term: nope, as it would most likely just be a re-implementation of `diart` (that internally relies on `pyannote.audio` models anyway).
* Long term: maybe.
